### ISSUE TITLE
Cell barcode calls do not match between hashing-based (demuxEM) and RNA-based in high background sample

### ISSUE THREAD

**Issue by heathergeiger** [broadinstitute/CellBender#92]
We are currently testing a nuclear hashing protocol. The barcode rank plot seems to indicate that background is very high. So, we thought it may be useful to give CellBender a try to remove background, and to better determine which cells are actually really cells.

Barcode rank plot is attached. One from CellBender, one drawn manually including only nuclei with at least 100 genes expressed.
![cellranger_barcode_rank_plot](https://user-images.githubusercontent.com/36860218/111175150-7fb6ec00-857e-11eb-8b9a-02389a389f85.png)
![manually_drawn_barcode_rank_plot_with_truncated_x_axis](https://user-images.githubusercontent.com/36860218/111175202-8a718100-857e-11eb-9157-3048c26b1835.png)

Based on this plot, I tried running CellBender with --expected-cells=20000. For --total-droplets-included, I tried both 50000 and 80000, with nearly identical results in terms of which cells were called as real cells (19538-19546 cells, with 19437 overlap between the two lists).

Since this is a cell hashing protocol, I also tried running with a raw h5 file resulting from CellRanger's feature barcoding module, thus including hashtag counts. This also resulted in a very similar list to the other runs (19658 cells, incl. 19420 three-way overlap with the total droplets 50k and 80k runs).

My concern is that CellBender is still missing some real cells. I also ran a piece of software called demuxEM (https://github.com/klarman-cell-observatory/demuxEM), which works on the hashing data to distinguish which cells are real or not. Using a relatively high cutoff on the RNA (min nUMI=1000, min nGene=700), then classifying cells meeting that threshold based on their hashtag profile as singlet/doublet/unknown (basically hashtag-negative), this software gives the following counts: 11741 doublet, 25854 singlet, 39983 unknown.

In other words, there are quite a few cells that are tagged as singlet or doublet according to the hashtag counts, with relatively high RNA levels, that are still being labeled as background. And including the hashtag counts in the input to CellBender does not change this.

Attached please find log files and diagnostic plots both for run with and without include HTO information (both with total droplets=50000).

[incl_HTO_run.log](https://github.com/broadinstitute/CellBender/files/6142498/incl_HTO_run.log)
[incl_HTO_run.pdf](https://github.com/broadinstitute/CellBender/files/6142499/incl_HTO_run.pdf)
[no_HTO_run.log](https://github.com/broadinstitute/CellBender/files/6142500/no_HTO_run.log)
[no_HTO_run.pdf](https://github.com/broadinstitute/CellBender/files/6142503/no_HTO_run.pdf)

List of cell barcodes from each of these:

[incl_HTO_barcodes.txt](https://github.com/broadinstitute/CellBender/files/6142550/incl_HTO_barcodes.txt)
[no_HTO_barcodes.txt](https://github.com/broadinstitute/CellBender/files/6142551/no_HTO_barcodes.txt)

Finally, please find attached demuxEM results in CSV format, as well as raw HTO count matrix.

[raw_HTO_counts.csv.gz](https://github.com/broadinstitute/CellBender/files/6142524/raw_HTO_counts.csv.gz)
[nuclear_hashing_test_demuxEM_results.csv.gz](https://github.com/broadinstitute/CellBender/files/6142538/nuclear_hashing_test_demuxEM_results.csv.gz)

The raw h5 file is too large to upload here (over 300Mb), but let me know if you have a way to upload that and I can include that as well.

My main question is, do you think some of the cells called as singlet by demuxEM that are not in the CellBender output are real cells? If so, how can I adjust the CellBender settings to be more sensitive?

**Comment by heathergeiger** [comment]
Oh one more thing. I did make this CSV file with the nGene and nUMI for all barcodes with at least one UMI if that helps. See attached.

[nuclear_hashing_test_nUMI_and_nGene_raw.csv.gz](https://github.com/broadinstitute/CellBender/files/6142729/nuclear_hashing_test_nUMI_and_nGene_raw.csv.gz)

**Comment by sjfleming** [comment]
Hi @heathergeiger !  Thanks for asking such an interesting question, this is a really great use-case, and I have not personally tried out a hashing protocol myself.

I also appreciate you including all this information, it makes it much easier for me to get an idea of what's going on.

The plot you included from the "no_HTO" CellBender output looks pretty close to what I'd expect if I were to guess cell/not based on just eyeballing the UMI curve.

![image](https://user-images.githubusercontent.com/10214815/113365944-d0bc3380-9325-11eb-8039-e68b22de1ef7.png)

I would definitely be willing to bet that the true number of cells is quite close to 20000, and that (11741 doublet, 25854 singlet) seems pretty high to me.  Doesn't that mean that about 37000 droplets contain at least one cell?  To me, that doesn't necessarily seem consistent with the UMI curve.

But if you disagree, let me know!
In short, I don't think that CellBender would budge all that much on the predicted number of cells based on varying parameters...
